trie/verkle.go

```diff
@@ -258,12 +258,22 @@ func (trie *VerkleTrie) Commit(_ bool) (common.Hash, *NodeSet, error) {
 	}
 
 	batch := trie.db.diskdb.NewBatch()
+	const keyPrefix = "flat-"
+	path := make([]byte, 0, len(keyPrefix)+32)
+	path = append(path, []byte(keyPrefix)...)
 	for _, node := range nodes {
-		if err := batch.Put(append([]byte("flat-"), node.Path...), node.SerializedBytes); err != nil {
+		path := append(path[:len(keyPrefix)], node.Path...)
```

Comment:
With the `path[:len(keyPrefix)]` we're "erasing" anything after `flat-`.

Reply:
are we sure that `batch.Put` will make a copy, though?

Reply:
[Looking at the LevelDB API](https://github.com/syndtr/goleveldb/blob/master/leveldb/batch.go#L119-L124):
> // Put appends 'put operation' of the given key/value pair to the batch.
// It is safe to modify the contents of the argument after Put returns but not
// before.

So, from my interpretation, should be safe.
Looking deeper in the implementation of that `.Put(..)` call, it does a [copy](https://github.com/syndtr/goleveldb/blob/master/leveldb/batch.go#L107) of the key, which seems aligned with the API comment.

Reply:
cool, thanks for checking